src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -172,80 +176,86 @@ bool CLeastAngleRegression::train_machine(CFeatures* data)
 	// main loop
 	//========================================
 	int32_t nloop=0;
-	while (m_num_active < max_active_allowed && max_corr > CMath::MACHINE_EPSILON && !stop_cond)
+	while (m_num_active < max_active_allowed && max_corr/n_vec > 1.192E-07 && !stop_cond)
 	{
 		// corr = X' * (y-mu) = - X'*mu + Xy
-		copy(Xy.begin(), Xy.end(), corr.begin());
-		cblas_dgemv(CblasColMajor, CblasTrans, n_vec, n_fea, -1,
-			X.matrix, n_vec, &mu[0], 1, 1, &corr[0], 1);
-
+		Map<VectorXd> map_corr(&corr[0], n_fea);
+		Map<VectorXd> map_mu(&mu[0], n_vec);
+		
+		map_corr = map_Xy - (map_X.transpose()*map_mu);
```

Comment:
again why not to use `map_Xr*map_mu` here instead of `map_X.transpose()*map_mu`?